src/Actions/OutputsDslQueryAction.php

```diff
@@ -96,4 +101,26 @@ private function mapCondition(array $condition): array
 
         return $parser->parse($field, $value);
     }
+
+    private function getMappings(): array
+    {
+        $indexName = config('rapidez.es_prefix').'_products_'.config('rapidez.store');
+        $esMappings = ElasticSearch::indices()->getMapping(['index' => $indexName]);
+        $mappings = data_get($esMappings, '*.mappings.properties', []);
+
+        return count($mappings) ? array_pop($mappings) : [];
+    }
+
+    private function getFieldMapping(string $attribute): string
+    {
+        if (! isset($this->mappings[$attribute])) {
+            return $attribute;
+        }
+
+        $mapping = $this->mappings[$attribute];
+
+        $keyword = $mapping['type'] === 'text' && $mapping['fields']['keyword']['type'] === 'keyword';
+
+        return $keyword ? $attribute.'.keyword' : $attribute;
+    }
```

Comment:
I feel like this function name does not represent what this function does. It's not returning a mapping, it's returning the name of the field that we should query for.

Reply:
Agreed!